src/dataflow/source/kafka.rs

```diff
@@ -351,12 +360,21 @@ fn update_partition_list(
     while i32::try_from(partitions.len()).unwrap() < expected_partition_count {
         let result = consumer.fetch_metadata(Some(&topic), Duration::from_secs(1));
         partitions = match &result {
-            Ok(meta) => match meta.topics().iter().find(|t| t.name() == topic) {
-                Some(topic) => {
-                    HashSet::from_iter(topic.partitions().iter().map(|x| x.id()).collect_vec())
-                }
-                None => HashSet::new(),
-            },
+            Ok(meta) => meta
+                .topics()
+                .iter()
+                .find(|t| t.name() == topic)
+                .map(|topic| {
```

Comment:
I don't quite grok why this code is changing. Does it have the same behavior as before, just in imperative style instead of with iterators?

Reply:
Yeah it has the same behavior, it was just a little weird to me to do the half-fluent/half-imperative control. I should have put it in the previous mild-refactor commit, which I will do now.

Reply:
oh, actually this is a partial state, too, originally I made more modifications to this, and didn't revert all the way, it's a nonsense refactor as is, will undo it.